core/src/main/scala/kafka/metrics/package.scala

```diff
@@ -0,0 +1,55 @@
+/**
+ *
+ *
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka
+
+import javax.management.{MalformedObjectNameException, ObjectName}
+import com.codahale.metrics._
+import com.codahale.metrics.jmx.{JmxReporter, ObjectNameFactory}
+
+import scala.collection.JavaConverters._
+
+package object metrics {
+
+  private[metrics] val kafkaMetricRegistry = new MetricRegistry()
```

Comment:
I wanted to bring attention to this proposal - https://cwiki.apache.org/confluence/display/KAFKA/KIP-544%3A+Make+metrics+exposed+via+JMX+configurable

which would introduce a separate Registry as well

Perhaps it would be best to get that KIP in first and then build this one on top for a more incremental and easier-to-digest approach

Reply:
Makes sense @stanislavkozlovski 
Then, I rebase the code.